R/plot.R

```diff
@@ -0,0 +1,157 @@
+#' Plot rolling case and death data with CFR estimates
+#'
+#' @description Plots the case and death time series used to calculate the CFR
+#' estimates in one panel and the resulting rolling naive and corrected CFR
+#' estimates in another panel.
+#'
+#' @param df_ncfr A data.frame containing the rolling naive CFR estimates, of
+#' the form returned by rolling_cfr.R
+#'
+#' @param df_ccfr A data.frame containing the rolling corrected CFR estimates,
+#' of the form returned by rolling_cfr.R
+#'
+#' @return A ggplot object of the case and death time series used to calculate
+#' the CFR estimates in one panel (top panel) and the resulting naive and
+#' corrected CFR estimates in another panel (bottom panel)
+#' @export
+#'
+#' @examples
+#' # Read in onset to death distribution for Ebola
+#' onset_to_death_ebola <- epiparameter::epidist("ebola", "onset_to_death")$pmf
+#'
+#' # Calculate rolling naive and corrected CFRs
+#' df_ncfr <- rolling_cfr(ebola1976, correct_for_delays = FALSE)
+#' df_ccfr <- rolling_cfr(ebola1976,
+#'   correct_for_delays = TRUE,
+#'   onset_to_death_ebola
+#' )
+#'
+#' # Plotting rolling estimates
+#' plot_data_and_cfr(df_ncfr, df_ccfr)
+#'
+plot_data_and_cfr <- function(df_ncfr, df_ccfr) {
+
+  # Some basic input checking
+  stopifnot(
+    "Naive rolling CFR estimates must have columns `cases` and `deaths`" =
+      (all(c("cases", "deaths", "date") %in% colnames(df_ncfr))),
+    "Corrected rolling CFR estimates must have columns `cases` and `deaths`" =
+      (all(c("cases", "deaths", "date") %in% colnames(df_ccfr)))
+  )
+
+  # set variables to NULL
+  value <- NULL
+  variable <- NULL
+  cfr_low <- NULL
+  cfr_high <- NULL
+  cfr_me <- NULL
+  type <- NULL
```

Comment:
We've had bad experience in socialmixr with this way of defining global variables. You should use `globalVariables()` instead or better, use the `.data` pronoun in your ggplot calls.

https://ggplot2.tidyverse.org/articles/ggplot2-in-packages.html

Reply:
Changed to `.data` in https://github.com/epiverse-trace/datadelay/pull/11/commits/a2daa07d06830f72171e85a4eae7a0fb4ab33af8.